arm-documentdb/2015-04-08/documentdb.json

```diff
@@ -0,0 +1,767 @@
+      "get": {
+        "operationId": "DatabaseAccounts_ListReadOnlyKeys",
+        "description": "Lists the read-only access keys for the specified database account.",
+        "parameters": [
+          {
+            "$ref": "#/parameters/subscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/resourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/accountNameParameter"
+          },
+          {
+            "$ref": "#/parameters/apiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "",
+            "schema": {
+              "$ref": "#/definitions/DatabaseAccountListReadOnlyKeysResult"
+            }
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.DocumentDB/databaseAccounts/{accountName}/regenerateKey?api-version={api-version}": {
+      "post": {
+        "operationId": "DatabaseAccounts_RegenerateKey",
+        "description": "Regenerates an access key for the specified database account.",
+        "parameters": [
+          {
+            "$ref": "#/parameters/subscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/resourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/accountNameParameter"
+          },
+          {
+            "$ref": "#/parameters/apiVersionParameter"
+          },
+          {
+            "name": "keyToRegenerate",
+            "in": "body",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/DatabaseAccountRegenerateKeyParameters"
+            },
+            "description": "Specifies name of the key which should be regenerated."
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "",
+            "schema": {
+              "$ref": "#/definitions/DatabaseAccountListKeysResult"
+            }
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/providers/Microsoft.DocumentDB/databaseAccountNames/{accountName}?api-version={api-version}": {
+      "head": {
+        "operationId": "DatabaseAccounts_CheckNameAvailability",
+        "description": "Checks that account name is valid and is not in use.",
+        "parameters": [
+          {
+            "$ref": "#/parameters/subscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/accountNameParameter"
+          },
+          {
+            "$ref": "#/parameters/apiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK. The account name is valid but already unavailable (already in use)."
+          },
+          "404": {
+            "description": "Not Found. The account name is available and valid."
+          },
+          "400": {
```

Comment:
This is a weird implementation of CheckNameAvailability. Usually services in Azure return a response with 200 status code. The response body has properties like { nameAvailable, reason, message }. Is it possible for the service to follow this pattern?